src/main/java/com/example/syncrowbackend/friendrequestroom/controller/PostController.java

```diff
@@ -0,0 +1,51 @@
+package com.example.syncrowbackend.friendrequestroom.controller;
+
+import com.example.syncrowbackend.friendrequestroom.dto.FriendRequestPostDto;
+import com.example.syncrowbackend.friendrequestroom.dto.PostRequestDto;
+import com.example.syncrowbackend.friendrequestroom.dto.PostDto;
+import com.example.syncrowbackend.friendrequestroom.enums.FriendRequestStatus;
+import com.example.syncrowbackend.friendrequestroom.service.PostService;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.data.web.PageableDefault;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("/api")
+@RequiredArgsConstructor
+public class PostController {
+
+    @Autowired
+    private final PostService postService;
+
+    @PostMapping("/posts/{kakaoId}")
+    public ResponseEntity<Long> createPost(@PathVariable String kakaoId, @Valid @RequestBody PostRequestDto postDto) {
+        return ResponseEntity.ok(postService.createPost(kakaoId, postDto));
+    }
```

Comment:
```suggestion
    @PostMapping("/posts")
    public ResponseEntity<Long> createPost(@Valid @RequestBody PostRequestDto postDto, @AuthenticationPrincipal UserDetailsImpl userDetails) {
        return ResponseEntity.ok(postService.createPost(userDetails.getUser(), postDto));
    }
```

스프링 시큐리티를 사용해 로그인한 유저 정보를 UserDetailsImpl 에 담아놨습니다 !
UserDetailsService 에서 DB에 저장된 유저를 가져오므로 `userDetails.getUser()` 로 User 객체를 가져올 수 있습니다 🙂

Reply:
와아! 그런 방법이 있었군요🫢 참고해서 바꿔보겠습니다!! 👍👍